src/main/java/de/tum/in/www1/artemis/domain/Result.java

```diff
@@ -483,14 +483,43 @@ public void filterSensitiveInformation() {
      * @param isBeforeDueDate if feedbacks marked with visibility 'after due date' should also be removed.
      */
     public void filterSensitiveFeedbacks(boolean isBeforeDueDate) {
+        filterSensitiveFeedbacks(isBeforeDueDate, participation.getExercise());
+    }
+
+    /**
+     * Removes all feedback details that should not be passed to the student.
+     *
+     * @param isBeforeDueDate if feedbacks marked with visibility 'after due date' should also be removed.
+     * @param exercise        the exercise related to this result. Used to determine if test case names should be removed.
+     */
+    public void filterSensitiveFeedbacks(boolean isBeforeDueDate, Exercise exercise) {
         feedbacks.removeIf(Feedback::isInvisible);
 
         if (isBeforeDueDate) {
             feedbacks.removeIf(Feedback::isAfterDueDate);
         }
 
-        // TODO: this is not good code!
```

Comment:
Stephan specifically added this some time ago regarding the change of the test case count and so on. Is it ok to remove this TODO now?